arrayfire/array.py

```diff
@@ -1185,20 +1185,35 @@ def to_list(self, row_major=False):
         ct_array, shape = self.to_ctype(row_major, True)
         return _ctype_to_lists(ct_array, len(shape) - 1, shape)
 
-    def __repr__(self):
+    def to_string(self):
```

Comment:
I understand your concerns, but the main reason of this PR is to make AF Python compatible with the popular PyCharm. With `__str__` we'll break PyCharm, with `to_string` we won't. If we open a ticket of this issue, as I know Jetbrains ticket processing system, two things might happen:

* they give us a reasonable explanation about why calling `str()` is necessary for PyCharm
* they categorize it as a bug, and it will get resolved within 2 years from now on

On the other hand I don't believe that the current behavior of array to string conversion is correct. Printing something is gotta be:

* a lightweight operation 
* that intended to show an object to user in a way that fits in a regular terminal window which is often 80*25 chars wide. 

In your neural network example, there is a subset of MNIST dataset in an array, which has a size of 28*28*10000. It's not that large, we often see this size of arrays on real datasets. If user calls `print` then he *accidentally* hit an operation that invokes:

* 7840000 * 4 bytes of device to host memory transfer
* 7840000 * 8 character printed to his terminal at once

It's the exact opposite that `print()` gotta be about, right?

So I strongly believe that method should be there of course, because it is good for showing small arrays, but it should not be invoked by any known standard Python interface *accidentally*, because those are designed to be lightweight, and IDEs like PyCharm correctly believes that invoking them internally won't be an issue.